Sources/ProjectDescription/Dependencies/CarthageDependencies.swift

```diff
@@ -1,76 +1,117 @@
 import Foundation
 
-/// Contains the description of external dependency that can by installed using Tuist.
-public enum Dependency: Codable, Equatable {
-    /// Origin of the Carthage dependency
-    public enum CarthageOrigin: Codable, Equatable {
-        /// Mimics `github` keyword from `Cartfile`. GitHub repositories (both GitHub.com and GitHub Enterprise).
-        case github(path: String)
-        /// Mimics `git` keyword from `Cartfile`. Other Git repositories.
-        case git(path: String)
-        /// Mimics `binary` keyword from `Cartfile`. Dependencies that are only available as compiled binary `.frameworks`.
-        case binary(path: String)
+/// Contains the description of dependency that can by installed using Carthage.
+public struct CarthageDependencies: Codable, Equatable {
+    /// List of depedencies that can by installed using Carthage.
+    public let dependencies: [Dependency]
+    /// List of platforms for which you want to install depedencies.
+    public let platforms: Set<Platform>
+    /// Indicates whether the Carthage produces XCFrameworks or regular frameworks.
```

Comment:
What about specifying that this is equivalent to pass `--use-xcframeworks` to Carthage?
Moreover we could also add that a version above the 0.37.0 is required